diff --git a/nursenextdoor/nursenextdoor/CommonTools/CommonDataModels/UIButtonDataModel.h b/nursenextdoor/nursenextdoor/CommonTools/CommonDataModels/UIButtonDataModel.h
index 02d8714..70f207d 100644
--- a/nursenextdoor/nursenextdoor/CommonTools/CommonDataModels/UIButtonDataModel.h
+++ b/nursenextdoor/nursenextdoor/CommonTools/CommonDataModels/UIButtonDataModel.h
@@ -36,7 +36,7 @@ typedef struct {
  Public properties
  */
 @property (retain, nonatomic) id<UIButtonDataModelControllerDelegate> delegate;
-@property (strong, nonatomic) NSString * titleString;
+
 @property (strong, nonatomic) UIButton * uiButtonReturnObject;
 @property (retain, nonatomic) id reciverClass;
 // This is the right size of the UIVIew that should be followed;
@@ -45,8 +45,19 @@ typedef struct {
 @property (nonatomic,assign)  CGFloat  modelFrameOriginey;
 @property (nonatomic,assign)  CGFloat  modelFrameSizeHeight;
 @property (nonatomic,assign)  CGFloat  modelFrameSizeWidth;
-@property (nonatomic,strong)  UIColor  *modelFrameBackGroundColor;
-@property (nonatomic,strong)  NSString *modelTitleString;
+@property (nonatomic,strong)  UIColor  *modelUIViewFrameBackGroundColor;
+
+
+
+
+/// Sized for the UIButton
+
+@property (nonatomic,assign)  CGFloat  modelUIButtonFrameOriginex;
+@property (nonatomic,assign)  CGFloat  modelUIButtonFrameOriginey;
+@property (nonatomic,assign)  CGFloat  modelUIButtonFrameSizeHeight;
+@property (nonatomic,assign)  CGFloat  modelUIButtonFrameSizeWidth;
+@property (nonatomic,strong)  UIColor  *modelUIButtonBackGroundColor;
+@property (nonatomic,strong)  NSString  *modelUIButtonTitle;
 
 
 
@@ -57,7 +68,7 @@ typedef struct {
  Date: July 24rd 2017
  README: When the button is initated, the UIBotton also correlated to an UIView.
  */
-@property (strong, nonatomic) UIView *uiViewUiButto;
+@property (strong, nonatomic) UIView *uiViewButtonDataModel;
 
 
 /*
diff --git a/nursenextdoor/nursenextdoor/CommonTools/CommonDataModels/UIButtonDataModel.m b/nursenextdoor/nursenextdoor/CommonTools/CommonDataModels/UIButtonDataModel.m
index e794afe..8a76716 100644
--- a/nursenextdoor/nursenextdoor/CommonTools/CommonDataModels/UIButtonDataModel.m
+++ b/nursenextdoor/nursenextdoor/CommonTools/CommonDataModels/UIButtonDataModel.m
@@ -41,7 +41,7 @@ -(instancetype) initWithDataModel: (UIButtonDataModel*) model{
     if (self) {
 
         _uiButtonReturnObject =[[UIButton alloc]initWithFrame:CGRectMake(0, 600 , 100, 100)];
-        [_uiButtonReturnObject setTitle:@"checkIn" forState:UIControlStateNormal];
+        [_uiButtonReturnObject setTitle:model.modelUIButtonTitle/*@"checkIn"*/ forState:UIControlStateNormal];
         [_uiButtonReturnObject setTitleColor:[UIColor greenColor] forState:UIControlStateNormal];
         [_uiButtonReturnObject addTarget:model.reciverClass action:@selector(callBackFunctionUIButtonDataModel:) forControlEvents:UIControlEventTouchUpInside];
         [_uiButtonReturnObject setBackgroundColor:[UIColor blueColor]];
@@ -61,8 +61,118 @@ -(instancetype) initWithDataModel: (UIButtonDataModel*) model{
 };
 
 
--(void) callBackFunctionUIButtonDataModel{
-    NSLog(@"test");
+- (instancetype)initWithXModel:(UIButtonDataModel*) model{
+    self = [super init];
+    if (self) {
+
+
+
+
+// CGRect Size for the UIVIew
+        // CGrect sizes for the Button itself
+        // Type for the Button
+        // CallBackClass which is NNDBarDataModel
+
+
+        // Build the UIView
+        UIButtonDataModel *localModel = [[UIButtonDataModel alloc]init];
+        localModel.reciverClass = model.reciverClass;
+        localModel.modelUIButtonTitle = model.modelUIButtonTitle;
+        localModel.uiViewButtonDataModel = [[UIView alloc]initWithFrame:CGRectMake(model.modelFrameOriginex, model.modelFrameOriginey, model.modelFrameSizeWidth, model.modelFrameSizeHeight)];
+        [localModel.uiViewButtonDataModel setBackgroundColor:model.modelUIViewFrameBackGroundColor];
+
+        // Build the UIButton
+   //     model.uiButtonReturnObject = [[UIButton alloc] initWithFrame:model.modelUiButtonFrameOrigineX];
+
+/*
+
+ 
+ // building the main uiview size 
+ @property (nonatomic,assign)  CGFloat  modelFrameOriginex;
+ @property (nonatomic,assign)  CGFloat  modelFrameOriginey;
+ @property (nonatomic,assign)  CGFloat  modelFrameSizeHeight;
+ @property (nonatomic,assign)  CGFloat  modelFrameSizeWidth;
+ @property (nonatomic,strong)  UIColor  *modelFrameBackGroundColor;
+ 
+ 
+
+        _uiButtonScheduledVisits = [[UIButtonDataModel alloc]initWithDataModel:builder2];
+        [_uiButtonScheduledVisits.uiButtonReturnObject setFrame:CGRectMake(100, 600 , 100, 100)];
+        // Set a pointer to the self class.
+        _uIViewNNDBar = self;
+
+
+        // test1
+        // built uiview for the button.
+        _uiButtonScheduledVisits.uiViewButtonDataModel = [[UIView alloc]initWithFrame:CGRectMake(_uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameOriginex,
+                                                                                         _uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameOriginey,
+                                                                                         _uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameSizeWidth,
+                                                                                         (_uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameSizeHeight - _uiViewSizesDatamodel.logoUiButtonFrameSizeHeight))];
+        // build a lable
+        UILabel *nextVisist= [[UILabel alloc] initWithFrame:CGRectMake(0,0, 500, 500)];
+        [nextVisist setText:@"This is the visit page "];
+        [_uiButtonScheduledVisits.uiViewButtonDataModel setBackgroundColor:[UIColor grayColor]];
+        
+        
+        [_uiButtonScheduledVisits.uiViewButtonDataModel insertSubview:nextVisist atIndex:0];
+        
+        [_uIViewNNDBar.uiVIewNNDBarView addSubview:_uiButtonScheduledVisits.uiButtonReturnObject];
+        
+        [_uIViewNNDBar.uiVIewNNDBarView addSubview:_uiButtonHome.uiButtonReturnObject];
+        
+        // End of the test#2
+*/
+
+
+
+
+    }
+    return self;
 }
 
+
+
+
+
+/*
+ 
+
+ // Repeat the same process for checkInScreen
+
+
+
+
+ UIButtonDataModel *builder2 = [[UIButtonDataModel alloc]init];
+ builder2.reciverClass = self;
+ builder2.modelUIButtonTitle =@"Visit";
+ _uiButtonScheduledVisits = [[UIButtonDataModel alloc]initWithDataModel:builder2];
+ [_uiButtonScheduledVisits.uiButtonReturnObject setFrame:CGRectMake(100, 600 , 100, 100)];
+ // Set a pointer to the self class.
+ _uIViewNNDBar = self;
+
+
+ // test1
+ // built uiview for the button.
+ _uiButtonScheduledVisits.uiViewButtonDataModel = [[UIView alloc]initWithFrame:CGRectMake(_uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameOriginex,
+ _uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameOriginey,
+ _uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameSizeWidth,
+ (_uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameSizeHeight - _uiViewSizesDatamodel.logoUiButtonFrameSizeHeight))];
+ // build a lable
+ UILabel *nextVisist= [[UILabel alloc] initWithFrame:CGRectMake(0,0, 500, 500)];
+ [nextVisist setText:@"This is the visit page "];
+ [_uiButtonScheduledVisits.uiViewButtonDataModel setBackgroundColor:[UIColor grayColor]];
+
+
+ [_uiButtonScheduledVisits.uiViewButtonDataModel insertSubview:nextVisist atIndex:0];
+
+ [_uIViewNNDBar.uiVIewNNDBarView addSubview:_uiButtonScheduledVisits.uiButtonReturnObject];
+
+ [_uIViewNNDBar.uiVIewNNDBarView addSubview:_uiButtonHome.uiButtonReturnObject];
+
+ // End of the test#2
+
+
+
+ */
+
 @end
diff --git a/nursenextdoor/nursenextdoor/SlidingFrameWork/Views/UIViewNNDBar.m b/nursenextdoor/nursenextdoor/SlidingFrameWork/Views/UIViewNNDBar.m
index 813cc02..988fce8 100644
--- a/nursenextdoor/nursenextdoor/SlidingFrameWork/Views/UIViewNNDBar.m
+++ b/nursenextdoor/nursenextdoor/SlidingFrameWork/Views/UIViewNNDBar.m
@@ -47,14 +47,76 @@ - (instancetype)initWithDataModel: (UIViewNNDBar*) uiViewNNDBarIncoming AndSizes
 
         UIButtonDataModel *builder = [[UIButtonDataModel alloc]init];
         builder.reciverClass = self;
+        builder.modelUIButtonTitle =@"Home";
 
         _uiButtonHome = [[UIButtonDataModel alloc]initWithDataModel:builder];
         // Set a pointer to the self class.
         _uIViewNNDBar = self;
 
 
+        // test1
+        // built uiview for the button.
+        _uiButtonHome.uiViewButtonDataModel = [[UIView alloc]initWithFrame:CGRectMake(_uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameOriginex,
+                                                                              _uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameOriginey,
+                                                                              _uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameSizeWidth,
+                                                                              (_uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameSizeHeight - _uiViewSizesDatamodel.logoUiButtonFrameSizeHeight))];
+        // build a lable
+        UILabel *welcome = [[UILabel alloc] initWithFrame:CGRectMake(0, 100, 400, 400)];
+        [welcome setText:@"This is the home page"];
+        [_uiButtonHome.uiViewButtonDataModel setBackgroundColor:[UIColor clearColor]];
+
+
+        [_uiButtonHome.uiViewButtonDataModel insertSubview:welcome atIndex:0];
+
+
+
+
+        // Repeat the same process for checkInScreen
+
+
+
+
+        UIButtonDataModel *builder2 = [[UIButtonDataModel alloc]init];
+        builder2.reciverClass = self;
+        builder2.modelUIButtonTitle =@"Visit";
+        _uiButtonScheduledVisits = [[UIButtonDataModel alloc]initWithDataModel:builder2];
+        [_uiButtonScheduledVisits.uiButtonReturnObject setFrame:CGRectMake(100, 600 , 100, 100)];
+        // Set a pointer to the self class.
+        _uIViewNNDBar = self;
+
+
+        // test1
+        // built uiview for the button.
+        _uiButtonScheduledVisits.uiViewButtonDataModel = [[UIView alloc]initWithFrame:CGRectMake(_uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameOriginex,
+                                                                              _uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameOriginey,
+                                                                              _uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameSizeWidth,
+                                                                              (_uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameSizeHeight - _uiViewSizesDatamodel.logoUiButtonFrameSizeHeight))];
+        // build a lable
+        UILabel *nextVisist= [[UILabel alloc] initWithFrame:CGRectMake(0,0, 500, 500)];
+        [nextVisist setText:@"This is the visit page "];
+        [_uiButtonScheduledVisits.uiViewButtonDataModel setBackgroundColor:[UIColor grayColor]];
+
+
+        [_uiButtonScheduledVisits.uiViewButtonDataModel insertSubview:nextVisist atIndex:0];
+
+        [_uIViewNNDBar.uiVIewNNDBarView addSubview:_uiButtonScheduledVisits.uiButtonReturnObject];
+
         [_uIViewNNDBar.uiVIewNNDBarView addSubview:_uiButtonHome.uiButtonReturnObject];
 
+        // End of the test#2
+
+
+
+
+
+
+
+        [_uIViewNNDBar.uiVIewNNDBarView addSubview:_uiButtonHome.uiButtonReturnObject];
+
+
+
+
+
         //[_uIViewNNDBar.uiVIewNNDBarView insertSubview:_uiButtonHome.uiButtonReturnObject atIndex:0 ];
 
         /*
@@ -108,6 +170,11 @@ -(void) callBackFunctionUIButtonDataModel:(UIButton*) sender{
     
 }
 
+
+
+
+
+
 - (void)updateUiViewAnimcation:(NSString *)animationID finished:(NSNumber *)finished context:(void *)context {
 
 
@@ -135,6 +202,9 @@ - (void)updateUiViewAnimcation:(NSString *)animationID finished:(NSNumber *)fini
                                                                                                  _uiViewSizesDatamodel.uiviewPermenantConnectionToSlidingUIViewModelFrameSizeHeight)];
 
                              [_uiviewPermenantConnectionToSlidingUIViewModel setBackgroundColor:[UIColor greenColor]];
+                             //[_uiviewPermenantConnectionToSlidingUIViewModel addSubview:_uiButtonHome.uiViewButtonDataModel];
+                             [_uiviewPermenantConnectionToSlidingUIViewModel addSubview:_uiButtonScheduledVisits.uiViewButtonDataModel];
+
                          }completion:^(BOOL finished){
                              NSLog(@"Face Up done");
                              
